src/github.com/getlantern/flashlight/config/config.go

```diff
@@ -273,21 +273,28 @@ func (cfg Config) cloudPollSleepTime() time.Duration {
 	return time.Duration((CloudConfigPollInterval.Nanoseconds() / 2) + rand.Int63n(CloudConfigPollInterval.Nanoseconds()))
 }
 
-func (cfg Config) fetchCloudConfig() ([]byte, error) {
+func (cfg Config) fetchCloudConfig() (bytes []byte, err error) {
 	log.Debugf("Fetching cloud config from: %s", cfg.CloudConfig)
-	// Try it unproxied first
-	bytes, err := cfg.doFetchCloudConfig("")
-	if err != nil && cfg.IsDownstream() {
-		// If that failed, try it proxied
-		bytes, err = cfg.doFetchCloudConfig(cfg.Addr)
+
+	if cfg.IsDownstream() {
+		// Clients must always proxy the request
+		if cfg.Addr == "" {
```

Comment:
IIUC we are not "polling" anyway, in the sense that no connections are attempted.  So the only effect is negligible CPU usage.

If an empty address was expected at all in normal usage, I'd probably make the change you suggest.  Since this is a defensive check, in principle I prefer to keep it close to where the sensitive actions (in this case, sending out requests to S3) are actually performed, so more possible causes for an empty address are caught.  So, if anything, my inclination woud be to move this into doFetchCloudConfig.

But I'm not convinced of that either, and considering that the importance of this check is dubious to begin with, I don't think it's worth holding this PR to pinpoint the perfect location for it.